packages/swapper/src/swappers/ThorchainSwapper/endpoints.ts

```diff
@@ -360,45 +359,46 @@ export const thorchainApi: SwapperApi = {
       const thorTxHash = txHash.replace(/^0x/, '')
 
       // not using monadic axios, this is intentional for simplicity in this non-monadic context
-      const { data } = await axios.get<ThornodeStatusResponse>(
-        `${config.REACT_APP_THORCHAIN_NODE_URL}/lcd/thorchain/tx/status/${thorTxHash}`,
-      )
-
-      if ('error' in data) {
+      const [{ data: txData }, { data: txStatusData }] = await Promise.all([
+        axios.get<ThornodeTxResponse>(
+          `${config.REACT_APP_THORCHAIN_NODE_URL}/lcd/thorchain/tx/${thorTxHash}`,
+        ),
+        axios.get<ThornodeStatusResponse>(
+          `${config.REACT_APP_THORCHAIN_NODE_URL}/lcd/thorchain/tx/status/${thorTxHash}`,
+        ),
+      ])
+
+      // We care about txStatusData errors because it drives all of the status logic.
+      if ('error' in txStatusData) {
```

Comment:
turbo nitpick: since we early return here, we could move from a `Promise.all()` to only fetching `txStatusData` (as we were doing before) and only fetching `txData` after this bit, to avoid one additional request in the case of errors